src/Presentation/TabsHtmlBuilder.php

```diff
@@ -55,13 +53,9 @@ private function parseQuery( string $searchQuery ): Query {
 	}
 
 	private function buildSettingsViewModel(): array {
-		if ( !$this->enableWikiConfig ) {
-			return [];
-		}
-
-		$title = Title::newFromText( WikibaseFacetedSearchExtension::CONFIG_PAGE_TITLE, NS_MEDIAWIKI );
+		$title = $this->titleFactory->newFromText( WikibaseFacetedSearchExtension::CONFIG_PAGE_TITLE, NS_MEDIAWIKI );
```

Comment:
I am not sure about having `TitleFactory` in this class, but we need `Title` to get the URL and the `PageIdentity` object needed for checking permissions.

Apart from that, we used `Title::newFromText` in multiple places to get the `Title` object for the configuration page. Should that be turned into a class?

Reply:
Using `TitleFactory` instead of `Title::newFromText` isn't hugely important if we do not make use of the polymorphic seam, such as via a test double. Then again, it is fairly easy to default to the OO approach of injecting `TitleFactory`